source/extensions/filters/common/ext_authz/check_request_utils.cc

```diff
@@ -118,13 +119,24 @@ void CheckRequestUtils::setHttpRequest(
   // Set request body.
   const Buffer::Instance* buffer = sdfc->decodingBuffer();
   if (max_request_bytes > 0 && buffer != nullptr) {
-    const uint64_t len = std::min(buffer->length(), max_request_bytes);
-    // std::unique_ptr<char[]> data(new char[len]);
+    const uint64_t length = std::min(buffer->length(), max_request_bytes);
     std::string data;
-    data.reserve(len);
-    buffer->copyOut(0, len, &data[0]);
+    data.reserve(length);
```

Comment:
Instead of `reserve` if you do `resize` you should be able to do copyOut to &data[0].

Reply:
Interesting! I was hitting a segfault. I think I could also initialize the string with the correct length, so it doesn't reallocate memory.